tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -1087,15 +1087,23 @@ class CudnnRnnDescriptor : public dnn::RnnDescriptor {
     // We can only reasonably expect the user to handle the subsequent failure
     // in profile mode, which is run with algorithms returned from
     // GetRnnAlgorithms() (which are non-default and explicitly set whether to
-    // use tensor ops).
-    if (RnnTensorOpMathEnabled() && algorithm_config.algorithm().has_value()) {
-      cudnnMathType_t math_type =
-          algorithm_config.algorithm()->tensor_ops_enabled()
-              ? CUDNN_TENSOR_OP_MATH
-              : CUDNN_DEFAULT_MATH;
+    // use tensor ops). CuDNN 7.2.1 fixed this issue
+    if (RnnTensorOpMathEnabled()) {
+      cudnnMathType_t math_type;
+      if (algorithm_config.algorithm().has_value()) {
+        math_type = algorithm_config.algorithm()->tensor_ops_enabled()
+                        ? CUDNN_TENSOR_OP_MATH
+                        : CUDNN_DEFAULT_MATH;
+      } else {
+#if CUDNN_VERSION >= 7201
+        math_type = CUDNN_TENSOR_OP_MATH;
```

Comment:
Just to double check: if tensor core is not available it'll still fallback to non-tensorcore operations right?

Reply:
Yes, after 7.2.1 the tensor_core_op_math is only a hint.

Reply:
In that case please document the behavior.

Reply:
This has already been doc'ed above. Someone has already mentioned why they didn't use the tensor core op as default one. And I added that  `CuDNN 7.2.1 fixed this issue`.

Reply:
I see, sorry I missed that.